src/CompiledMatcher.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of composer/semver.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\Semver;
+
+use Composer\Semver\Constraint\ConstraintInterface;
+
+/**
+ * Helper class to evaluate constraint by compiling and reusing the code to evaluate
+ */
+class CompiledMatcher
+{
+    private $cacheDir;
+
+    private $checked;
+
+    public function __construct($cacheDir = null)
+    {
+        $this->cacheDir = $cacheDir ? $cacheDir : \sys_get_temp_dir();
+    }
+
+    /**
+     * Evaluates the expression: $constraint match $operator $version
+     *
+     * @param ConstraintInterface $constraint
+     * @param int                 $operator
+     * @param string              $version
+     *
+     * @return mixed
+     */
+    public function match(ConstraintInterface $constraint, $operator, $version)
+    {
+        $cacheKey = $operator.$constraint;
+        if (!isset($this->checked[$cacheKey])) {
+            $sha = \sha1($cacheKey);
+            $file = $this->cacheDir.'/constraint_'.$sha;
```

Comment:
I would add a `.php` extension on the file

Reply:
thanks